internal/transformer/classes/generators/dockerfilegenerators/dotnet/webappdockerfilegenerator.go

```diff
@@ -0,0 +1,154 @@
+func (t *WinWebAppDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *WinWebAppDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *WinWebAppDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory: %s", err)
+		return nil, nil, err
+	}
+	appName := ""
+	ports := make([]string, 0)
+	for _, de := range dirEntries {
+		if filepath.Ext(de.Name()) != dotnet.CsSln {
+			continue
+		}
+		csProjPaths, err := parseSolutionFile(filepath.Join(dir, de.Name()))
+		if err != nil {
+			logrus.Errorf("%s", err)
+			continue
+		}
+
+		if len(csProjPaths) == 0 {
+			logrus.Errorf("No projects available for the solution: %s", de.Name())
+			continue
+		}
+
+		for _, csPath := range csProjPaths {
+			projPath := filepath.Join(strings.TrimSpace(dir), strings.TrimSpace(csPath))
+			byteValue, err := ioutil.ReadFile(projPath)
+			if err != nil {
+				logrus.Errorf("Could not read the project file: %s", err)
+				continue
+			}
+
+			configuration := dotnet.CSProj{}
+			err = xml.Unmarshal(byteValue, &configuration)
+			if err != nil {
+				logrus.Errorf("Could not parse the project file: %s", err)
+				continue
+			}
+
+			if configuration.PropertyGroup == nil ||
+				configuration.PropertyGroup.TargetFrameworkVersion == "" ||
+				!dotnet.FourXPattern.MatchString(configuration.PropertyGroup.TargetFrameworkVersion) {
+				continue
+			}
+
+			isWebProj, err := isWeb(configuration)
+			if err != nil {
+				logrus.Errorf("%s", err)
+				continue
+			}
+			if !isWebProj {
+				continue
+			}
+
+			isSLProj, err := isSilverlight(configuration)
+			if err != nil {
+				logrus.Errorf("%s", err)
+				continue
+			}
+			if isSLProj {
+				continue
+			}
+
+			appName = strings.TrimSuffix(filepath.Base(de.Name()), filepath.Ext(de.Name()))
+		}
+
+		// Exit soon of after the solution file is found
+		break
+	}
+
+	if appName == "" {
+		return nil, nil, nil
+	}
+
+	namedServices = map[string]transformertypes.ServicePlan{
+		appName: []transformertypes.TransformerPlan{{
+			Mode:              t.Config.Spec.Mode,
+			ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			Paths: map[string][]string{
+				artifacts.ProjectPathPathType: {dir},
+			},
+			Configs: map[string]interface{}{
+				artifacts.DockerfileTemplateConfigConfigType: map[string]interface{}{
+					"Ports":            ports,
```

Comment:
Might be better to move the port extraction to transform step

Reply:
done